scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -428,8 +428,13 @@ def dijkstra(csgraph, directed=True, indices=None,
         will decrease computation time by aborting calculations between pairs
         that are separated by a distance > limit. For such pairs, the distance
         will be equal to np.inf (i.e., not connected).
-
         .. versionadded:: 0.14.0
+    multi_target: bool, optional
+        If False (default), independently find the shortest path to every index
+        in indices.  If True, find the shortest path to any index in indices and
+        the resulting distance matrix with be Kx1 vector, where K is the len(indices)
```

Comment:
### Readability

I find these docs a bit difficult to decode. How about:
```
    multi_target: bool, optional
￼        If False (default), for every node in the graph, find the shortest path to every node in indices.
        If True, for every node in the graph, find the shortest path to any of the nodes in indices (which
        can be substantially faster).
...

    dist_matrix : ndarray, shape ([n_indices, ]n_nodes,)
        The matrix of distances between graph nodes. If multi_target=False, dist_matrix
        has shape (n_indices, n_nodes) and dist_matrix[i, j] gives the shortest distance
        from point i to point j along the graph. If mult_target=True, dist_matrix has
        shape (n_nodes,) and contains the shortest path from each node to any
        of the nodes in indices.
    predecessors : ndarray, shape ([n_indices, ]n_nodes,)
        If multi_target=False, this has shape (n_indices, n_nodes),
        otherwise it has shape (n_nodes,).
        ...
```
... tweaked to be PEP8 compatible, of course.

### Completeness

One thing I find missing in this implementation is an output of the set of indices that yielded the minimal distances. Maybe this should be added to the end of `Returns` and returned as:
```
    min_indices : ndarray, shape (n_nodes,)
        Only returned if multi_target=True. Contains the node numbers
        that yielded the minimum distances.
```
In other words, in many cases you may want to know which nodes gave you the min distances, and there should be almost no cost to outputting this value as well as the distances themselves.

### Naming

I don't find `multi_target` to be the clearest naming. What it seems to be doing is more a minimum across all `indices`, so a name like `min_only=True` to compute/return the minimum of the shortest distances probably makes more sense than `multi_target=True`.

Reply:
I like the doc changes, i've incorporated them locally, will push after i do tests.  I also like min_only better than multi_target, so i've adopted that. 

In terms of the output, this would require some deeper modification of the C components in the Fibonacci heap to allow for tracking which target node was the reason for each of shortest paths on each source node reached.  I think this is an extra optimization, as the information is embedded in the pred network and comes with an added memory cost.  I could implement the 'expensive' version of this easily by just taking on a loop that follows along the pred path after getting them.  I've added that expensive version as a test now.

Reply:
Sounds good, does it hurt performance much?

Reply:
so for any particular node, you start at that node's index in pred, and then you go to the index at that spot, hopping your way along till you encounter a -9999 entry.  You should only get there if it's impossible to reach a target (in which case you make 0 hops), or when you end up on one of the target indices you passed in (in which you make a number of hops equal to the path from your starting node to the closest target).

Reply:
my new test actually implements this logic now to do the check you suggested.

Reply:
It makes sense this would be true for the output you get in the `indices=None, min_index=False` case, but it doesn't look like the `min_index=True` case has a `predecessors` matrix of the correct shape to traverse this way. Am I missing something?

Reply:
Okay if the tests pass (which it looks like they did) I must just be missing something about how `pred` works. So I'll trust the result there :)

Reply:
i have now implemented this tracking so you get back what index was closest to each target when doing min_only.